kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java

```diff
@@ -37,26 +41,26 @@ public CachingChunkManager(
       SearchContext searchContext,
       String s3Bucket,
       String dataDirectoryPrefix,
-      int slotCountPerInstance) {
+      int slotCountPerInstance)
+      throws Exception {
     this.meterRegistry = registry;
     this.metadataStore = metadataStore;
     this.blobFs = blobFs;
     this.searchContext = searchContext;
     this.s3Bucket = s3Bucket;
     this.dataDirectoryPrefix = dataDirectoryPrefix;
     this.slotCountPerInstance = slotCountPerInstance;
+
+    replicaMetadataStore = new ReplicaMetadataStore(metadataStore, false);
+    snapshotMetadataStore = new SnapshotMetadataStore(metadataStore, false);
+    searchMetadataStore = new SearchMetadataStore(metadataStore, false);
+    cacheSlotMetadataStore = new CacheSlotMetadataStore(metadataStore, false);
   }
 
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting caching chunk manager");
 
-    ReplicaMetadataStore replicaMetadataStore = new ReplicaMetadataStore(metadataStore, false);
```

Comment:
We should keep these initializations in startup, or move all of the codebase to initializing these in the constructor. All other classes initialize temporary metadata stores in the `startup` (ie RecoveryService, IndexingChunkManager, etc).

The reason these are in the startup is an artifact of when the metadatastore could have been "un-started" when these classes were initialized. This is no-longer the case though, as we've moved the metadata startup prior to service initialization .

Reply:
Leaving them in startup for now.